### PR TITLE
Simplify logging in `pytsql.py`

### PR DESCRIPTION
**Pull request by svengiegerich**
**Motivation**: When parsing multiple files, the package produces quite some logging. These entries are not telling much and create overhead.

**Changes**: Remove two logging entries - which are more helpful for debugging (in my view). Besides, I propose to formulate the source of another one more explicitly.

### CODE REVIEW

**Review by ivergara — APPROVED** [review]
Wait for others' opinions, but for me looks sensible.

File: src/pytsql/tsql.py [hunk 1]
```diff
@@ -154,8 +154,6 @@ def _split(code: str, isolate_top_level_statements: bool = True) -> List[str]:
             " and is preventing the use of the faster C++ parser."
         )
 
-    logging.info("Started SQL script parsing")
```
Comment (SimeonStoykovQC) [inline src/pytsql/tsql.py:157]: Some more general suggestions:
1. We could also log those as debug statements
2. We could use a dedicated named logger for `pytsql` like `logger = logging.getLogger("pytsql")`, and then people also have the opportunity to disable all `pytsql` logs

That said, I don't find those statements that helpful, so I am also fine with removing them.

**Review by SimeonStoykovQC — APPROVED** [review]
Looks good already, but here are some thoughts

File: src/pytsql/tsql.py (hunk 1 above)
Comment (ivergara) [reply src/pytsql/tsql.py:157]: As we discussed, I'd turn them into `debug` even when they're not too helpful and give the logger a name. Improving the content of the logging itself can be done in a separate effort.

File: src/pytsql/tsql.py [hunk 2]
```diff
@@ -154,7 +154,7 @@ def _split(code: str, isolate_top_level_statements: bool = True) -> List[str]:
...
+    logging.debug("Started SQL script parsing")
```
Comment (SimeonStoykovQC) [inline src/pytsql/tsql.py:157]: All `logging.*` statements should most probably be turned into `logger.*`, where `logger = logging.getLogger("pytsql")`

File: environment.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ dependencies:
   - sphinx
   - sphinx_rtd_theme
   - sphinxcontrib-apidoc
-  - sqlalchemy
+  - sqlalchemy==1.4.*
```
Comment (SimeonStoykovQC) [inline environment.yml:16]: Just to make sure: this is already pinned in the resulting package, so this is to make sure the development environment is consistent with it, right?

https://github.com/Quantco/pytsql/blob/e8bedc516a7cad9de908274687e494b00d1fb5eb/pyproject.toml#L39

**Review by SimeonStoykovQC — APPROVED** [review]
Looks good to me!

Idea: could you add a test where we try to disable the `pytsql` logger? It might not work well with `pytest`'s log capturing, so I would not put too much effort if it doesn't work straight away...

File: environment.yml (hunk 3 above)
Comment (SimeonStoykovQC) [reply environment.yml:16]: > Also, I propose to pin sqlalchemy to 1.4.* for now. As the integration tests are failing with 2.0 at the moment -

Then let's do `sqlalchemy<2.0` to be more explicit about the intent.
Indeed, `pytsql` is not using [SQLAlchemy 2.0 style](https://docs.sqlalchemy.org/en/14/changelog/migration_20.html) :/

E.g. it should be using `sqlalchemy.text` for raw SQL

https://github.com/Quantco/pytsql/blob/6fdf20492630b2860cca2301e0ce4e9b764d4e50/src/pytsql/tsql.py#L224-L228

Comment (SimeonStoykovQC) [reply environment.yml:16]: > should we than either (i) pin pyproject more explicitly within this PR as well, or (ii) separate these changes in a distinct PR?

Either is fine by me.

File: pyproject.toml [hunk 4]
```diff
@@ -36,7 +36,7 @@ dynamic = ["version"]
 requires-python = ">=3.7.0"
 
 dependencies = [
-    "sqlalchemy >=1.4",
+    "sqlalchemy <2.0",
```
Comment (SimeonStoykovQC) [inline pyproject.toml:39]: I was hoping we could still have both requirements?

File: src/pytsql/tsql.py (hunk 1 above)
Comment (ivergara) [reply src/pytsql/tsql.py:157]: Good question. Not sure. My guess is that you don't need it in the `__init__.py`.

File: tests/integration/test_logging.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import logging
+
+from pytsql import executes
+
+
+def test_deactivation_of_logger(engine, caplog):
```
[the remark this answers is not in the document]
Comment (ivergara) [reply tests/integration/test_logging.py:6]: To me looks good. Just don't see the need to undo the disabling and setLevel at the end.

File: tests/integration/test_logging.py [hunk 6]
```diff
@@ -14,4 +14,3 @@ def test_deactivation_of_logger(engine, caplog):
     assert len(caplog.records) == 0
 
     logger.disabled = False
```
Comment (ivergara) [inline tests/integration/test_logging.py:16]:
```suggestion

```
I'd say you don't need this either. And if you do, it'd mean there is something wrong in the testing suite.

File: tests/integration/test_logging.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ivergara) [reply tests/integration/test_logging.py:6]: Then my case stands and the tests were not independent of each other. Add the disabling again then.

**Pull request merged**